src/WebApiContrib.Core.Formatter.Csv/CsvInputFormatter.cs

```diff
@@ -0,0 +1,141 @@
+﻿#region copyright
+// Copyright 2016 WebApiContrib
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Net.Http.Headers;
+
+namespace WebApiContrib.Core.Formatter.Csv
+{
+    /// <summary>
+    /// ContentType: text/csv
+    /// </summary>
+    public class CsvInputFormatter : InputFormatter
+    {
+        private readonly CsvFormatterOptions _options;
+
+        public CsvInputFormatter(CsvFormatterOptions csvFormatterOptions)
+        {
+            if (csvFormatterOptions == null)
+            {
+                throw new ArgumentNullException(nameof(csvFormatterOptions));
+            }
+
+            _options = csvFormatterOptions;
+        }
+
+        public override Task<InputFormatterResult> ReadRequestBodyAsync(InputFormatterContext context)
+        {
+            var type = context.ModelType;
+            var request = context.HttpContext.Request;
+            MediaTypeHeaderValue requestContentType = null;
+            MediaTypeHeaderValue.TryParse(request.ContentType, out requestContentType);
+
+
+            var result = readStream(type, request.Body);
+            return InputFormatterResult.SuccessAsync(result);
+        }
+
+        public override bool CanRead(InputFormatterContext context)
+        {
+            var type = context.ModelType;
+            if (type == null)
+                throw new ArgumentNullException("type");
+
+            return isTypeOfIEnumerable(type);
+        }
+
+        private bool isTypeOfIEnumerable(Type type)
+        {
+
+            foreach (Type interfaceType in type.GetInterfaces())
+            {
+
+                if (interfaceType == typeof(IList))

```

Comment:
`IEnumerable` is more generic, makes more sense IMHO. Also, this method is repeated in both formatters so would be a good candidate for an extension method